File: top-interview-questions-medium/Array and Strings/set-matrix-zeroes/73.py
class Solution:
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        firstRowZero, firstColZero = False, False
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if (matrix[i][j] == 0 and i != 0 and j != 0 ):
                    matrix[i][0] = 0
                    matrix[0][j] = 0
                elif matrix[i][j] == 0:
                    firstRowZero = True if i == 0 else firstRowZero
                    firstColZero = True if j == 0 else firstColZero

        for i in range(1, len(matrix)):
            for j in range(1, len(matrix[i])):
                if (matrix[0][j] == 0 or matrix[i][0] == 0): matrix[i][j] = 0

        if firstColZero:
            for i in range(len(matrix)): matrix[i][0] = 0

        if firstRowZero:
            for j in range(len(matrix[0])): matrix[0][j] = 0

File: top-interview-questions-medium/Array and Strings/set-matrix-zeroes/73.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
  let firstRowZero = false, firstColZero = false
  for (let i = 0; i < matrix.length; i++) {
    for (let j = 0; j < matrix[i].length; j++) {
      if (matrix[i][j] === 0 && i && j) {
        matrix[i][0] = 0
        matrix[0][j] = 0
      } else if (matrix[i][j] === 0) {
        firstRowZero = i === 0 ? true : firstRowZero
        firstColZero = j === 0 ? true : firstColZero
      }
    }
  }
  for (let i = 1; i < matrix.length; i++) {
    for (let j = 1; j < matrix[0].length; j++) {
      if (matrix[0][j] == 0 || matrix[i][0] == 0) matrix[i][j] = 0
    }
  }
  for (let i = 0; firstColZero && i < matrix.length; i++) {
    matrix[i][0] = 0
  }
  for (let j = 0; firstRowZero && j < matrix[0].length; j++) {
    matrix[0][j] = 0
  }
}
